src/Processors/Executors/ExecutionThreadContext.cpp

```diff
@@ -60,7 +60,7 @@ static void executeJob(ExecutingGraph::Node * node, ReadProgressCallback * read_
                     read_progress_callback->addTotalBytes(read_progress->counters.total_bytes);
 
                 if (!read_progress_callback->onProgress(read_progress->counters.read_rows, read_progress->counters.read_bytes, read_progress->limits))
-                    node->processor->cancel();
+                    return false;
```

Comment:
Is it okay that we are not calling `cancel()` here? Is it possible that we will introduce a delay during cancel procedure this way?